.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,32 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/typescript-node
+{
+	"name": "Game",
+	"build": {
+		"dockerfile": "Dockerfile"
```

Comment:
Just out of curiosity would it be similar to use one of the images specified here instead? https://hub.docker.com/_/microsoft-vscode-devcontainers

Like `mcr.microsoft.com/vscode/devcontainers/javascript-node`

Reply:
You can definitely use one of those. The MCR images are substantially larger, so they take longer to download and start. On a local machine, the download time is mitigated if you have other environments using the MCR images. Natively, those image have more tools integrated (if that matters) and can be easier to configure. The tradeoff is they are less secure, so you wouldn't consider using the image in production (if that is a consideration).

The vendor image is smaller and more focused. Reducing the dependencies (and bundled tools) increases security, but can make customizing the image a bit tougher. As a result, you're developing on an image that could also be used to host the production application. You'll notice below that updating to Node 18.10 for the platform eliminates 3 known vulnerabilities.

Changing the base is minimal effort if you prefer something else. All of them support macOS (ARM & Intel) and PCs. For comparison:

| Image                                                                            | BaseOS | Size      | Snyk Scan Vulnerabilities |
| -------------------------------------------------- | ------- | ---------| --------------------- |
| mcr.microsoft.com ... javascript-node:18-bullseye  | Debian | 1.07 GB   | 456 (4 crit, 56 high) |
| mcr.microsoft.com ... typescript-node:18-bullseye | Debian | 1.26 GB   | 391 (4 crit, 56 high) |
| node:18.3.0-alpine3.16                                                | Alpine  | 169.7 MB | 3 (1 crit, 1 high)        |
| node:18.10.0-alpine3.16                                              | Alpine  | 164.5 MB | 0                                |